server.go

```diff
@@ -202,20 +188,63 @@ func getRequestBody(input string) (*http.Request, error) {
 	inputs = append(inputs, data)
 	reqBody := RequestBody{inputs}
 	fmt.Print(reqBody)
-	//reqBodyBytes, err := json.Marshal(reqBody)
-	//if err != nil {
-		//return nil, err
-	//}
 
 	b := new(bytes.Buffer)
-  err := json.NewEncoder(b).Encode(reqBody)
+	err := json.NewEncoder(b).Encode(reqBody)
 	req, err := http.NewRequest("POST", clarifaiApi, b)
 	fmt.Print(req)
 	if err != nil {
 		return nil, err
 	}
 
-	req.Header.Add("Authorization", `Bearer ThzzqYyVARtJdLbTQDwfRpa1FOk8w6`)
+	req.Header.Add("Authorization", `Bearer ` + clarifai_token.Token)
 	req.Header.Add("Content-Type", `application/json`)
 	return req, nil
 }
+
+func getAccesToken() AccessToken {
+	CLARIFAI_ID := os.Getenv("CLARIFAI_CLIENT_ID")
+	CLARIFAI_SECRET_KEY := os.Getenv("CLARIFAI_SECRET_KEY")
+
+	responseData := authRequest(CLARIFAI_ID, CLARIFAI_SECRET_KEY)
+	var token AccessToken
+	rawIn := json.RawMessage(responseData)
+
+	bytes, err := rawIn.MarshalJSON()
+	if err != nil {
+		fmt.Print("\nERROR Parsing authentication response")
+		panic(err)
+	}
+
+	err = json.Unmarshal(bytes, &token)
+	if err != nil {
+		fmt.Print("\nERROR Parsing authentication json string")
+		panic(err)
+	}
+	fmt.Print("\nRESULTS: ")
+	fmt.Print(token.Token)
+	fmt.Print(token.ExpiresIn)
+
+	return token
+}
+
+func authRequest(CLIENT_ID string, SECRET_KEY string) []byte {
+
+	resp, err := http.PostForm(clarifyAuth, url.Values{"grant_type": {"client_credentials"},
+		"client_id": { CLIENT_ID },
+		"client_secret": { SECRET_KEY } })
+
+	fmt.Print("\nAuthentication Request constructed and sent")
+
+	if err != nil {
+		panic(err)
```

Comment:
I'm ok with using panic for now. But if we want our project to be production quality, we probably want to avoid using panic. A better way will be throw an error_code with an error_msg. Then the client side is expected to handle it